<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TaskManager</class>
 <widget class="QWidget" name="TaskManager">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Task Manager</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">

   <!-- Top Row Inputs -->
   <item>
    <layout class="QHBoxLayout" name="inputLayout" stretch="2,1,1,1,1,1">
     <item>
      <widget class="QLineEdit" name="title_input">
       <property name="placeholderText">
        <string>Task Title</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDateEdit" name="date_picker">
       <property name="calendarPopup">
        <bool>true</bool>
       </property>
       <property name="displayFormat">
        <string>MM/dd/yyyy</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="priority_input">
       <item><property name="text"><string>Low</string></property></item>
       <item><property name="text"><string>Medium</string></property></item>
       <item><property name="text"><string>High</string></property></item>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="tags_input">
       <property name="placeholderText">
        <string>Tags</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="completed_input">
       <property name="text">
        <string>Completed</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="theme_toggle_button">
       <property name="text">
        <string>Dark Mode ðŸŒ™</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>

   <!-- Action Buttons -->
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item><widget class="QPushButton" name="add_button"><property name="text"><string>Add Task</string></property></widget></item>
     <item><widget class="QPushButton" name="edit_button"><property name="text"><string>Edit Task</string></property></widget></item>
     <item><widget class="QPushButton" name="delete_button"><property name="text"><string>Delete Task</string></property></widget></item>
     <item><widget class="QPushButton" name="save_button"><property name="text"><string>Save Tasks</string></property></widget></item>
     <item><widget class="QPushButton" name="load_button"><property name="text"><string>Load Tasks</string></property></widget></item>
    </layout>
   </item>

   <!-- Search + Smart Sort -->
   <item>
    <layout class="QHBoxLayout" name="searchLayout">
     <item>
      <widget class="QLineEdit" name="search_input">
       <property name="placeholderText">
        <string>Search...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="filter_button">
       <property name="text">
        <string>Filter Tasks</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="smart_sort_checkbox">
       <property name="text">
        <string>Enable Smart Sort</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>

   <!-- Task Table -->
   <item>
    <widget class="QTableWidget" name="table">
     <column><property name="text"><string>Title</string></property></column>
     <column><property name="text"><string>Due Date</string></property></column>
     <column><property name="text"><string>Priority</string></property></column>
     <column><property name="text"><string>Tags</string></property></column>
     <column><property name="text"><string>Completed</string></property></column>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>